Sources/TuistKit/Cache/CacheController.swift

```diff
@@ -188,4 +198,33 @@ final class CacheController: CacheControlling {
 
         _ = try cache.store(hash: hash, paths: FileHandler.shared.glob(outputDirectory, glob: "*")).toBlocking().last()
     }
+
+    fileprivate func buildAndCacheBundle(path: AbsolutePath,
+                                         target: GraphTarget,
+                                         configuration: String,
+                                         hash: String) throws
+    {
+        let outputDirectory = try FileHandler.shared.temporaryDirectory()
+        defer {
+            try? FileHandler.shared.delete(outputDirectory)
+        }
+
+        if path.extension == "xcworkspace" {
+            try bundleArtifactBuilder.build(
+                workspacePath: path,
```

Comment:
if you didn't have two different methods as I suggested above, this could also be done in a more compact manner by creating a variable and just passing it to the `build` function after the if-else

Reply:
As I said it requires to change protocol signature.

Reply:
might be worth doing a refactor of the protocol then - but I'll leave it up to you, it's not necessary 🙂